Feat/auth

mhoo999 opened the pull request:
session 기반 인증 설계를 진행했습니다.

mhoo999 committed:
Type (Auth) : Session 기반 인증/인가 구현

### signup() 메서드 구현
- PasswordEncoder를 통해 비밀번호 암호화
- AuthService 에서 생성된 유저의 URI 를 반환하여, AuthController 에서 201 Created 를 반환
- 유효하지 않은 조건을 핸들링하기 위해 InvalidRequestException 추가
- 유저타입을 Enum으로 관리하기 위해 Usertype 추가
- Users entity, UserRepository 추가

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### login() 메서드 구현
- HttpSession 사용하여 token 정보 저장
- token 정보 반환

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### LoginFilter 구현

### 메서드 테스트
- /api/v1/auth/signup 정상
- /api/v1/auth/login 정상
  - session 반환

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 테스트 코드 작성
- AuthControllerTest
- PasswordEncoderTest

### @EnableJpaAuditing 분리
- JpaAuditingConfiguration.java 로 분리함

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 미사용 코드 삭제

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### token 반환 과정 삭제
- 추후 JWT 도입 시 추가할 예정

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 미사용 코드 삭제

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 미사용 코드 삭제

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/controller/AuthController.java:
```diff
@@ -0,0 +1,42 @@
+                .buildAndExpand(userId)
+                .toUri();
+
+        return ResponseEntity.created(location).body("회원가입이 완료되었습니다.");
```
body값은 지워주세요.
클라이언트에서 별도로 만들수 있거나
재사용이 안되는 문구는 없애는 편이 좋습니다.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/dto/request/LoginRequestDto.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Schema(description = "이메일", example = "sparta@sparta.com")
+    @NotBlank
+    @Email(message = "유효한 이메일 형식이 아닙니다.")
```
@Email 은 좋은 어노테이션이 아닙니다.
@Pattern 로 바꿔주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/dto/request/LoginRequestDto.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Schema(description = "비밀번호")
+    @NotBlank
+    @Pattern(regexp = "^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[\\W_]).{8,}$", message = "비밀번호는 최소 8글자 이상이어야 하며, 대소문자 포함 영문 + 숫자 + 특수문자를 최소 1글자씩 포함해야 합니다.")
```
message 에 들어가는 값을 interface -> final 로 만들어서 지정을 해주세요.
enum클래스는 message값으로 들어가지 못하니 참고해주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/dto/request/SignupRequestDto.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Schema(description = "이메일", example = "sparta@sparta.com")
+    @NotBlank
+    @Email(message = "유효한 이메일 형식이 아닙니다.")
```
위와 같은 이유입니다.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/dto/request/SignupRequestDto.java:
```diff
@@ -0,0 +1,45 @@
+
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
```
AllArgsConstructor 의 사용은 자제해주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/service/AuthService.java:
```diff
@@ -0,0 +1,60 @@
+import xyz.tomorrowlearncamp.outsourcing.global.exception.InvalidRequestException;
+
+@Service
+@RequiredArgsConstructor
```
@Validated 넣어주세요.
Controller를 제외하는 곳에서 @Valid를 사용하는 곳은 @Validated를 넣어주어야합니다.
https://velog.io/@baekgom/Valid%EC%99%80-Validated 참고해주세요.

mixedsider commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/service/AuthService.java:
```diff
@@ -0,0 +1,60 @@
+
+        session.setAttribute("LOGIN_USER", user.getId());
+
+        return "로그인 되었습니다.";
```
void 로 바꿔주세요. 위와 같은 이유입니다.

mixedsider reviewed (changes requested):
review: 코드 검토 완료

3uomlkh commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/domain/user/entity/Users.java:
```diff
@@ -0,0 +1,35 @@
+@Entity
+@NoArgsConstructor
+@Table(name = "users")
+public class Users extends BaseEntity {
```
Entity 이름을 코드 컨벤션에 맞게 작성해주시면 좋을 것 같습니다.

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 피드백 내용 적용
- Controller method의 response에서는 클라이언트에서 별도로 만들 수 있거나, 재사용이 안되는 문구는 지양
- `@Email` 어노테이션 사용 지양, `@Pattern` 으로 대체
    - 사용되는 message는 interface로 만들어서 재사용할 수 있도록 사용하는 것이 좋음
- Controller를 제외한 Bean에서는 `@Validated` 어노테이션을 넣어줘야 동작함

PS : 테스트 코드 작성중

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 사용하지 않는 코드 수정

mixedsider reviewed (approved):
review: 코드 검토 완료

3uomlkh commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/domain/user/entity/User.java:
```diff
@@ -0,0 +1,35 @@
+@Entity
+@NoArgsConstructor
+@Table(name = "users")
+public class User extends BaseEntity {
```
저희 팀의 코드 컨벤션은 `(도메인)Entity` 이기 때문에 이에 맞춰서 작성해주시면 좋을 것 같습니다.

3uomlkh commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/dto/request/ValidationMessages.java:
```diff
@@ -0,0 +1,8 @@
+package xyz.tomorrowlearncamp.outsourcing.auth.dto.request;
+
+public interface ValidationMessages {
```
상수를 관리해야 한다면 인터페이스보다는 final class 혹은 enum class로 하는 것도 괜찮을 것 같습니다.

3uomlkh commented on src/main/java/xyz/tomorrowlearncamp/outsourcing/auth/service/AuthService.java:
```diff
@@ -0,0 +1,60 @@
+        User user = userRepository.findByEmail(dto.getEmail()).orElseThrow(()-> new InvalidRequestException("이메일 또는 비밀번호가 일치하지 않습니다."));
+
+        if (!passwordEncoder.matches(dto.getPassword(), user.getPassword())) {
+            throw new InvalidRequestException("이메일 또는 비밀번호가 일치하지 않습니다.");
```
중복되는 에러 메세지 부분을 위에서 관리해주신 것처럼 따로 빼서 관리하면 좋을 것 같습니다.

3uomlkh reviewed (commented):
review: 크게 문제되는 부분은 없어보여서 조금씩 리팩토링 해가면 될 것 같습니다. Entity 이름만 컨벤션에 맞게 수정 부탁드립니다. 고생하셨습니다!

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 피드백 내용 적용

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 피드백 내용 적용

mhoo999 committed:
feat(Auth) : Session 기반 인증/인가 구현

### 피드백 내용 적용
